build_daemon/lib/data/server_log.dart

```diff
@@ -2,16 +2,94 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+import 'package:built_collection/built_collection.dart';
 import 'package:built_value/built_value.dart';
 import 'package:built_value/serializer.dart';
+import 'package:logging/logging.dart' as logging;
 
 part 'server_log.g.dart';
 
+/// Logging levels, these have a 1:1 mapping with the levels from
+/// `package:logging`.
+class Level extends EnumClass with Comparable<Level> {
+  static Serializer<Level> get serializer => _$levelSerializer;
+
+  static const Level FINEST = _$finest;
```

Comment:
why all caps?

Reply:
The name is derived from this, and we want it to line up with the old names (which were upper case). The alternative would be upper casing it during log output etc, which we would end up doing all the time.

Reply:
now that we have the build daemon for a programmatic API we don't need to worry as much about keeping identical log output. We can pick different labels.

I think it would be good to try something like https://github.com/dart-lang/build/pull/1165 again